modules/lima/main.nf

```diff
@@ -0,0 +1,50 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process LIMA {
+    tag "$meta.id"
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::lima=2.2.0" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/lima:2.2.0--h9ee0642_0"
+    } else {
+        container "quay.io/biocontainers/lima:2.2.0--h9ee0642_0"
+    }
+
+    input:
+    tuple val(meta), path(ccs)
+    path(primers)
+
+    output:
+    tuple val(meta), path("*.bam"), emit: bam
+    path("*.bam.pbi"),              emit: pbi
+    path("*.xml"),                  emit: xml
+    path("*.json"),                 emit: json
+    path("*.clips"),                emit: clips
+    path("*.counts"),               emit: counts
+    path("*.guess"),                emit: guess
+    path("*.report"),               emit: report
+    path("*.summary"),              emit: summary
+    path("*.version.txt"),          emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    def lima_out = ccs.toString().replaceAll(/bam$/, 'fl.bam')
```

Comment:
I think this (https://github.com/nf-core/modules/blob/97b803a8a7ed1a454cd85bd604a49637effca787/modules/bwamem2/mem/main.nf#L32) pattern would be more suitable here.

Reply:
It may also be helpful to support the other input types:
```
lima [options] <INPUT.bam|xml|fa|fq|gz> <BARCODES.fa> <OUTPUT.bam|xml|fa|fq|gz>
```
More specifically, does file ending logic need to be added based on input? Are any files optional given different input from BAM?